tests/php/RestApi/StoreApi/Controllers/CartShippingRates.php

```diff
@@ -168,13 +170,15 @@ public function test_get_item_schema() {
 	 * Test conversion of cart item to rest response.
 	 */
 	public function test_prepare_item_for_response() {
-		$controller = new \Automattic\WooCommerce\Blocks\RestApi\StoreApi\Controllers\CartShippingRates();
-		$packages   = wc()->shipping->calculate_shipping( wc()->cart->get_shipping_packages() );
-		$response   = $controller->prepare_item_for_response( current( $packages ), [] );
-		$data       = $response->get_data();
+		$controller           = new \Automattic\WooCommerce\Blocks\RestApi\StoreApi\Controllers\CartShippingRates();
+		$packages             = wc()->shipping->calculate_shipping( wc()->cart->get_shipping_packages() );
+		$packages[0]['index'] = 0;
```

Comment:
I'm curious, why is this being set?

Reply:
[By default](https://github.com/woocommerce/woocommerce/blob/master/includes/wc-cart-functions.php#L231), WooCommerce names each package as _Shipping 1_, _Shipping 2_, etc. I wanted to keep feature-parity and implemented this in [`CartShippingRateSchema.php`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1806/files#diff-a2a9064ef82874022e94727f82dbe981R227). In order to accomplish this, we need to know the index of each package, so it's added in [`CartShippingRates.php`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1806/files#diff-e84e70f2afe35e8aa7c42cc62c6028afR125). Tests needed to be updated accordingly.

Hope that makes sense. :slightly_smiling_face:

Reply:
So does this mean by default `$packages` here does not have an index key on the first element in the `$packages` array?  I'm wary about tests that have to have a forced value like this.

Reply:
> So does this mean by default `$packages` here does not have an index key on the first element in the `$packages` array?

None of the `$packages` returned by `wc()->shipping->calculate_shipping( wc()->cart->get_shipping_packages() );` will have an `index` key. We add that `key` in [`CartShippingRates.php`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1806/files#diff-e84e70f2afe35e8aa7c42cc62c6028afR125).

> I'm wary about tests that have to have a forced value like this.

I have spent some time thinking about how to refactor the code relative to this test but always ended up in a dead end. The point is that we need to know the package `index` inside [`get_item_response()`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1806/files#diff-a2a9064ef82874022e94727f82dbe981R201), so we can get the correct package name. Unfortunately, I couldn't see any way of doing that other than saving it as a property in `$package`. That means we need to update the `$package` shape also in tests, and that's the `$packages[0]['index'] = 0;` line. :slightly_smiling_face:

Reply:
Thanks for looking into this. Since it appears to do what is intended, let's roll with this for now. Maybe Mike could review this too in case he has ideas, it's still a bit puzzling to me and seems like a lot of unnecessary logic (but I'm trusting you here). Does this suggest that maybe `index` should be an array key set by these core functions?

Side note, I see in `wc-cart-functions.php` (in woo core) that there is an identical filter in use `woocommerce_shipping_package_name`. Is that a filter you are replicating here? 

It looks like travis build is failing, once you get that fixed, you can make the call on whether to merge or not 👍

Reply:
> Thanks for looking into this. Since it appears to do what is intended, let's roll with this for now. Maybe Mike could review this too in case he has ideas, it's still a bit puzzling to me and seems like a lot of unnecessary logic (but I'm trusting you here). Does this suggest that maybe `index` should be an array key set by these core functions?

Happy if @mikejolley can take a look and see if I missed a better way of doing this. :slightly_smiling_face: 

The issue is that we can't change the signature of `get_item_response()`,  so it can only get one parameter (`$package`) + `$request`. The only way I could find to have access to the array key of the `$package` inside there, was to store it as a `$package` property.

> Side note, I see in `wc-cart-functions.php` (in woo core) that there is an identical filter in use `woocommerce_shipping_package_name`. Is that a filter you are replicating here? 

Yes, that filter is used by extensions to give a name to the packages. I tried to keep it so we would not break that feature.

> It looks like travis build is failing, once you get that fixed, you can make the call on whether to merge or not +1

Those tests are failing in `master` too, so I think they should be fixed in another PR separate from this one.